nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/iot/GetAWSIoT.java

```diff
@@ -0,0 +1,153 @@
+})
+public class GetAWSIoT extends AbstractAWSIoTProcessor {
+
+    public static final List<PropertyDescriptor> properties = Collections.unmodifiableList(
+            Arrays.asList(
+                    PROP_QOS,
+                    PROP_TOPIC,
+                    PROP_ENDPOINT,
+                    PROP_KEEPALIVE,
+                    PROP_CLIENT,
+                    AWS_CREDENTIALS_PROVIDER_SERVICE,
+                    REGION));
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return Collections.singleton(REL_SUCCESS);
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        // init to build up mqtt connection over web-sockets
+        init(context);
+        if (mqttClient != null && mqttClient.isConnected()) {
+            try {
+                // subscribe to topic with configured qos in order to start receiving messages
+                mqttClient.subscribe(awsTopic, awsQos);
+            } catch (MqttException e) {
+                getLogger().error("Error while subscribing to topic " + awsTopic + " with client-id " + mqttClient.getClientId() + " caused by " + e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final List messageList = new LinkedList();
+        // check if connection is about to terminate
+        if (isConnectionAboutToExpire()) {
+            MqttWebSocketAsyncClient _mqttClient = null;
+            try {
+                // before subscribing to the topic with new connection first unsubscribe
+                // old connection from same topic if subscription is set to QoS 0
+                if (awsQos == 0) mqttClient.unsubscribe(awsTopic);
+                // establish a second connection
+                _mqttClient = connect(context);
+                // now subscribe to topic with new connection
+                _mqttClient.subscribe(awsTopic, awsQos);
+                // between re-subscription and disconnect from old connection
+                // QoS=0 subscription eventually lose some messages
+                // QoS=1 subscription eventually receive some messages twice
+                // now terminate old connection
+                mqttClient.disconnect();
+            } catch (MqttException e) {
+                getLogger().error("Error while renewing connection with client " + mqttClient.getClientId() + " caused by " + e.getMessage());
+            } finally {
+                // grab messages left over from old connection
+                mqttClient.getAwsQueuedMqttMessages().drainTo(messageList);
+                // now set the new connection as the default connection
+                if (_mqttClient != null) mqttClient = _mqttClient;
```

Comment:
Line 111 (_mqttClient.subscribe) can throw an MqttException which would lead to _mqttClient being initialized and connected but not subscribed. Then on this line it would replace mqttClient with _mqttClient. There should be a check to make sure the it successfully subscribed.